### PR TITLE
Simplify generation code

### PR DESCRIPTION
**Pull request by derekhiggins**
The post_process_gpt3_response function takes a lot of params that we don't use and returns a format we need to convert. Because of this we're looping on
datastructures for no reason. Remove it to simplify generation code.

Fixes: #485

### CODE REVIEW

**Comment by russellb** [comment]
Triggered an e2e job run: https://github.com/instructlab/instructlab/actions/runs/8974117468

**Comment by nathan-weinberg** [comment]
Job failed on this error
```bash
Run git fetch origin pull/931/head:pr-931
From https://github.com/instructlab/instructlab
 * [new ref]         refs/pull/931/head -> pr-931
error: The following untracked working tree files would be overwritten by checkout:
	src/instructlab/schema/v1/compositional_skills.json
	src/instructlab/schema/v1/knowledge.json
	src/instructlab/schema/v1/version.json
Please move or remove them before you switch branches.
Aborting
Error: Process completed with exit code 1.
```

**Comment by russellb** [comment]
OK -- that failure probably just means the PR needs to be rebased.

**Comment by russellb** [comment]
@mergifyio rebase

**Comment by russellb** [comment]
Another run post-rebase -- https://github.com/instructlab/instructlab/actions/runs/8976938629

**Comment by russellb** [comment]
@mergifyio rebase

**Comment by hickeyma** [comment]
@Mergifyio rebase

**Review by hickeyma — DISMISSED** [review]
Thanks for pushing this refactor @derekhiggins. It seems to improves the generation performance and is runs when tested manually.

I would like @xukai92 to take a look to see if I missed anything when reviewing the removed batch code.

**Comment by leseb** [comment]
@derekhiggins could you please rebase this code? Thanks!

@xukai92 could you please take another look? It looks like we are getting close but would like your review to finalize it. Thanks!

File: src/instructlab/generator/generate_data.py [hunk 1]
```diff
@@ -371,7 +357,6 @@ def generate_data(
     num_cpus: Optional[int] = None,
     num_instructions_to_generate: Optional[int] = None,
     num_prompt_instructions=2,
-    request_batch_size=5,
```
Comment (xukai92) [inline src/instructlab/generator/generate_data.py:374]: does this simplification actually changed this to (and fixed it as) `request_batch_size=1` effectively?

File: src/instructlab/generator/generate_data.py [hunk 2]
```diff
@@ -183,9 +182,7 @@ def writeline2file(logfile, line):
 def post_process_gpt3_response(num_prompt_instructions, response, discarded_file):
     if response is None:
         return [], 0
-    raw_instructions = (
-        f"* Task {num_prompt_instructions + 1}\n" + response.message.content
-    )
+    raw_instructions = f"* Task {num_prompt_instructions + 1}\n" + response
```
Comment (xukai92) [inline src/instructlab/generator/generate_data.py:185]: i would rename this variable to be `response_content` to avoid confusion

**Comment by russellb** [comment]
Apologies for the trouble, but could you repost this to the new repo for this code?  https://github.com/instructlab/sdg

We're in the middle of the transition of moving this code to the new repo, and I'd like to minimize further changes made here.

**Comment by nathan-weinberg** [comment]
Since SDG code has moved to a different repo, I'm going to close this

**Pull request closed without merging**